.ci/test_r_package.sh

```diff
@@ -69,10 +69,10 @@ packages="c('data.table', 'jsonlite', 'Matrix', 'R6', 'testthat')"
 if [[ $OS_NAME == "macos" ]]; then
     packages+=", type = 'binary'"
 fi
-Rscript --vanilla -e "install.packages(${packages}, repos = '${CRAN_MIRROR}', lib = '${R_LIB_PATH}')" || exit -1
+Rscript --vanilla -e "install.packages(${packages}, repos = '${CRAN_MIRROR}', lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'))" || exit -1
```

Comment:
@jameslamb 
Just curious: what is the default set of options?

Reply:
This set is the default set. From `?install.packages`

> The default, NA, means c("Depends", "Imports", "LinkingTo").

Those are the minimum ones needed to load a package. In case you're not familiar with them:

* `Depends`: Has to be present to install your package. Will be loaded whenever your package is loaded.
* `Imports`: Has to be present to install your package. Will only be loaded when your package calls code from them.
* `LinkingTo`: Has to be present to install your package. These are usually header-only files that your code links to, e.g. the [BH](https://cran.r-project.org/web/packages/BH/index.html) package for Boost.

This commit just makes that explicit.

Reply:
Got it, thanks a lot for the detailed explanation!